crates/hir-ty/src/lower.rs

```diff
@@ -837,9 +856,9 @@ impl<'a> TyLoweringContext<'a> {
                 fill_self_params();
             }
             let expected_num = if generic_args.has_self_type {
-                self_params + type_params + const_params
+                self_params + type_params + const_params + lifetime_params
             } else {
-                type_params + const_params
+                type_params + const_params + lifetime_params
             };
```

Comment:
This is used after the `.filter(|arg| !matches!(arg, GenericArg::Lifetime(_)))` on line 868, so I don't think it should account for lifetimes?

Reply:
I thought about it too, but I didn't change since it's used in take and take only returns what it has available even in the case of bigger number, that's why I didn't change it. Also enforces that `expected_num` includes lifetimes too. This is just my point of view. I could remove it if you want.